apps/mobile/src/components/widgets/collectibles/collectibles.layout.tsx

```diff
@@ -0,0 +1,81 @@
+import { NonFungibleCryptoAssetInfo } from '@leather.io/models';
+import { Box, CollectibleCard } from '@leather.io/ui/native';
+import { isDefined, isEmptyString } from '@leather.io/utils';
+
+import { CollectibleCaption } from './collectible-caption';
+
+interface CollectiblesLayoutProps {
+  collectibles: NonFungibleCryptoAssetInfo[];
+  view?: 'widget' | 'gallery';
+}
+
+function filterCollectibles(collectible: NonFungibleCryptoAssetInfo) {
+  switch (collectible.protocol) {
+    case 'inscription':
+      return !isEmptyString(collectible.src) && isDefined(collectible.src);
+    case 'sip9':
+      return (
+        !isEmptyString(collectible.cachedImage) &&
+        isDefined(collectible.cachedImage) &&
+        // BNS - Archive is not a valid collectible
+        collectible.name !== 'BNS - Archive'
+      );
+    default:
+      return false;
+  }
+}
+
+export function CollectiblesLayout({ collectibles, view = 'gallery' }: CollectiblesLayoutProps) {
+  const displayLimit = 9;
+  const thumbnailSize = view === 'widget' ? 200 : 164;
+  return (
+    <Box flexDirection="row" alignItems="center" paddingHorizontal="5" gap="4" flexWrap="wrap">
+      {collectibles.filter(filterCollectibles).map((collectible, index) => {
+        if (view === 'widget' && index >= displayLimit) return null;
```

Comment:
Adding this limit on the `widget` on `Home` as I don't think we should render all of them there. 

I'll check with design
![Screenshot 2025-03-26 at 15 25 31](https://github.com/user-attachments/assets/c4b3b120-c8ad-408c-9776-8a1f54ffb570)